src/components/featureHighlight/index.js

```diff
@@ -136,56 +145,82 @@ class FeatureHighlight extends BaseComponent {
           });
         }, 0);
       }
     } else {
       const frame = props.highlightFrame;
       if (frame) {
-        this.setState({
-          targetPosition: {left: frame.x, top: frame.y, width: frame.width, height: frame.height},
-        });
+        this.targetPosition = {left: frame.x, top: frame.y, width: frame.width, height: frame.height};
+        this.setContentPosition();
       }
     }
   }
 
-  getContentPositionStyle() {
-    const {highlightFrame, minimumRectSize, innerPadding} = this.props;
-    const {targetPosition, contentViewHeight} = this.state;
-    const {top, height} = targetPosition || {};
-    const screenVerticalCenter = Constants.screenHeight / 2;
-    const targetCenter = top + (height / 2);
-    const isAlignedTop = targetCenter > screenVerticalCenter;
-    let topPosition = isAlignedTop ? top - contentViewHeight : top + height;
-    if (!highlightFrame && !isAlignedTop) {
-      const minRectHeight = minimumRectSize.height;
-      const isUnderMin = height >= minRectHeight;
-      topPosition = isUnderMin ? topPosition + innerPadding : targetCenter + (minRectHeight / 2) + (innerPadding / 2);
-    }
-    if (topPosition < 0 || topPosition + contentViewHeight > Constants.screenHeight) {
-      console.warn('Content is too long and might appear off screen. ' +
-        'Please adjust the message length for better results.');
+  getContentPosition() {
+    if (this.didLayout) {
+      const {highlightFrame, minimumRectSize, innerPadding} = this.props;
+      const {top, height} = this.targetPosition;
+      const screenVerticalCenter = Constants.screenHeight / 2;
+      const targetCenter = top + (height / 2);
+      const isAlignedTop = targetCenter > screenVerticalCenter;
+      let topPosition = isAlignedTop ? top - this.contentHeight : top + height;
+      if (!highlightFrame && !isAlignedTop) {
+        const minRectHeight = minimumRectSize.height;
+        const isUnderMin = height >= minRectHeight;
+        topPosition = isUnderMin ? topPosition + innerPadding : targetCenter + (minRectHeight / 2) + (innerPadding / 2);
+      }
+      if (topPosition < 0 || topPosition + this.contentHeight > Constants.screenHeight) {
+        console.warn('Content is too long and might appear off screen. ' +
+          'Please adjust the message length for better results.');
+      }
+      return topPosition;
     }
-    return {top: topPosition};
+  }
+
+  setContentPosition() {
+    const top = this.getContentPosition();
+    this.setState({contentViewStyle: top});
+    this.animate(1);
   }
 
   // This method will be called more than once in case of layout change!
   getComponentDimensions(event) {
-    const height = event.nativeEvent.layout.height;
-    this.setState({contentViewHeight: height});
+    this.didLayout = true;
+    this.contentHeight = event.nativeEvent.layout.height;
+    if (this.targetPosition !== undefined) {
+      this.setContentPosition();
+    }
+  }
+
+  onCustomPress = () => {
```

Comment:
I would change this name also, "custom" is too general and we might have all kind of clicks in that screen.